In [4]:
# 1 Proyecto de modelo de análisis de datos en Python
# Alumno: Ángel Tomás Moreno Senén
# Curso de especialización programación con Python
# Curso escolar 2025/2026
# IES Severo Ochoa

In [5]:
# 2 El primer paso, importar pandas
import numpy as np
import pandas as pd


In [6]:
# 3 Subir los archivos
train_df = pd.read_csv("train.csv")
test_df = pd.read_csv("test.csv")


In [7]:
# 4 Comprobar que se han subido Train
print ("¿Ha subido el Train?")
train_df.head()


¿Ha subido el Train?


,id,ANYO_CONSTRUCCION,CODIGO_POSTAL,USO_EDIFICIO,MUNICIPIO,PROVINCIA,TIPO_REGISTRO,FICHERO_ORIGEN,CONSUMO_EP_VALOR
0,23809,2018.0,3178.0,VIVIENDA/S UNIFAMILIAR,BENIJÓFAR,ALICANTE,NUEVO,NaN,47
1,34002,2007.0,3410.0,VIVIENDA INDIVIDUAL,BIAR,ALICANTE,EXISTE,certificacion-energetica-de-edificios-en-alica...,165
2,40936,1996.0,3010.0,NaN,ALICANTE/ALACANT,ALICANTE,EXISTE,certificacion-energetica-de-edificios-en-alica...,89
3,41393,1975.0,46115.0,VIVIENDA INDIVIDUAL,ALFARA DEL PATRIARCA,VALENCIA,EXISTE,certificacion-energetica-de-edificios-en-valen...,184
4,8708,2002.0,NaN,VIVIENDA INDIVIDUAL,ROJALES,ALICANTE,EXISTE,certificacion-energetica-de-edificios-en-alica...,296


In [8]:
# 5 Comprobar que se han subido Test
print("¿Ha subido el Test?")
test_df.head()

¿Ha subido el Test?


,id,ANYO_CONSTRUCCION,CODIGO_POSTAL,USO_EDIFICIO,MUNICIPIO,PROVINCIA,TIPO_REGISTRO,FICHERO_ORIGEN
0,33554,1988.0,46680,VIVIENDA/S UNIFAMILIAR,ALGEMESÍ,VALENCIA,EXISTE,CERTIFICACION-ENERGETICA-DE-EDIFICIOS-EN-VALEN...
1,9428,2001.0,3177,VIVIENDA/S UNIFAMILIAR,SAN FULGENCIO,ALICANTE,EXISTE,CERTIFICACION-ENERGETICA-DE-EDIFICIOS-EN-ALICA...
2,200,1950.0,46006,VIVIENDA/S UNIFAMILIAR,VALENCIA,VALENCIA,EXISTE,CERTIFICACION-ENERGETICA-DE-EDIFICIOS-EN-VALEN...
3,12448,1992.0,3780,VIVIENDA INDIVIDUAL,PEGO,ALICANTE,EXISTE,CERTIFICACION-ENERGETICA-DE-EDIFICIOS-EN-ALICA...
4,39490,1950.0,46006,VIVIENDA INDIVIDUAL,VALENCIA,VALENCIA,EXISTE,CERTIFICACION-ENERGETICA-DE-EDIFICIOS-EN-VALEN...


In [9]:
# 6 A ver que lleva train
# LA VARIABLE A PREDECIR es CONSUMO_EP_VALOR, Y ES: valor numérico continuo y muy disperso y relacionado con el consumo energético.
# El resto de columnas: variables predictoras.
train_df.head()
train_df.info()
train_df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38250 entries, 0 to 38249
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 38250 non-null  int64 
 1   ANYO_CONSTRUCCION  36430 non-null  object
 2   CODIGO_POSTAL      36255 non-null  object
 3   USO_EDIFICIO       36281 non-null  object
 4   MUNICIPIO          36295 non-null  object
 5   PROVINCIA          36248 non-null  object
 6   TIPO_REGISTRO      36275 non-null  object
 7   FICHERO_ORIGEN     36447 non-null  object
 8   CONSUMO_EP_VALOR   38250 non-null  int64 
dtypes: int64(2), object(7)
memory usage: 2.6+ MB


,id,CONSUMO_EP_VALOR
count,38250.000000,38250.000000
mean,25023.821778,174.084078
std,14420.280457,172.779020
min,1.000000,1.000000
25%,12537.250000,119.000000
50%,24978.000000,161.000000
75%,37528.750000,213.000000
max,50000.000000,22479.000000


# HASTA AQUÍ VA.......... BIEEN

In [10]:
# 7 Verificación de valores nulos
print(train_df.isnull().sum())

id                      0
ANYO_CONSTRUCCION    1820
CODIGO_POSTAL        1995
USO_EDIFICIO         1969
MUNICIPIO            1955
PROVINCIA            2002
TIPO_REGISTRO        1975
FICHERO_ORIGEN       1803
CONSUMO_EP_VALOR        0
dtype: int64


Esto aquí y ahora da lo siguiente:
id                      0  
ANYO_CONSTRUCCION    1820  
CODIGO_POSTAL        1995  
USO_EDIFICIO         1969  
MUNICIPIO            1955  
PROVINCIA            2002  
TIPO_REGISTRO        1975  
FICHERO_ORIGEN       1803  
CONSUMO_EP_VALOR        0  
dtype: int64


LA VARIABLE A PREDECIR es CONSUMO_EP_VALOR, Y ES: valor numérico continuo y muy disperso y relacionado con el consumo energético.

El resto de columnas: variables predictoras.

In [11]:
# 8 Porcentaje de nulos
porcentaje_nulos = (train_df.isnull().sum() / len(train_df)) * 100
print(porcentaje_nulos.sort_values(ascending=False))

PROVINCIA            5.233987
CODIGO_POSTAL        5.215686
TIPO_REGISTRO        5.163399
USO_EDIFICIO         5.147712
MUNICIPIO            5.111111
ANYO_CONSTRUCCION    4.758170
FICHERO_ORIGEN       4.713725
id                   0.000000
CONSUMO_EP_VALOR     0.000000
dtype: float64


In [12]:
# 9 Verificación de filas duplicadas

duplicados = train_df.duplicated().sum()
print(f"Número de filas duplicadas: {duplicados}")

Número de filas duplicadas: 750


In [13]:
# 10 Eliminación de filas duplicadas

print(f"Filas antes de eliminar duplicados: {len(train_df)}")

train_df = train_df.drop_duplicates()
# ... para ver el resultado
print(f"Filas después de eliminar duplicados: {len(train_df)}")

# Me sale que 38250 - 37500 = 750 eliminadas ¡OK!

Filas antes de eliminar duplicados: 38250
Filas después de eliminar duplicados: 37500


In [14]:
# 11 Explorar valores únicos de columnas categóricas

print(train_df["USO_EDIFICIO"].unique())
print(train_df["TIPO_REGISTRO"].unique())
print(train_df["FICHERO_ORIGEN"].unique())
print(train_df["USO_EDIFICIO"].value_counts())

# A ver qué sale

['VIVIENDA/S UNIFAMILIAR' 'VIVIENDA INDIVIDUAL' nan '???'
 'LOCAL COMERCIAL' 'sin dato' 'OTROS' 'RESTAURANTE' 'Sin Dato' 'OFICINAS'
 '----' 'valencia' 'EDIFICIO DE VIVIENDAS (BLOQUE COMPLETO)' 'VALÈNCIA'
 'DESCONOCIDO' '  VALENCIA  ' 'HOSPITAL' 'EDIFICIO COMERCIAL'
 'CENTRO DE ENSEÑANZA' 'HOTEL' 'INSTALACIONES DEPORTIVAS']
['NUEVO' 'EXISTE' '  VALENCIA  ' nan 'Sin Dato' 'valencia' 'VALÈNCIA'
 '----' 'DESCONOCIDO' 'sin dato' '???' 'REFORMA']
[nan 'certificacion-energetica-de-edificios-en-alicante.csv'
 'certificacion-energetica-de-edificios-en-valencia.csv'
 'certificacion-energetica-de-edificios-en-castellon.csv']
USO_EDIFICIO
VIVIENDA INDIVIDUAL                        23409
VIVIENDA/S UNIFAMILIAR                      8674
LOCAL COMERCIAL                             1650
EDIFICIO DE VIVIENDAS (BLOQUE COMPLETO)      468
OFICINAS                                     317
OTROS                                        215
----                                          85
Sin Dato              

In [15]:
# 12 Limpieza de categorías: QUITAR ESPACIOS Y PONER EN MAYÚSCULAS

train_df["USO_EDIFICIO"] = train_df["USO_EDIFICIO"].str.strip().str.upper()

train_df["TIPO_REGISTRO"] = train_df["TIPO_REGISTRO"].str.strip().str.upper()

# y una miradita
display(train_df["USO_EDIFICIO"])
display(train_df["TIPO_REGISTRO"])

,USO_EDIFICIO
0,VIVIENDA/S UNIFAMILIAR
1,VIVIENDA INDIVIDUAL
2,NaN
3,VIVIENDA INDIVIDUAL
4,VIVIENDA INDIVIDUAL
...,...
38244,OTROS
38245,VIVIENDA INDIVIDUAL
38246,VIVIENDA INDIVIDUAL
38248,VIVIENDA INDIVIDUAL


,TIPO_REGISTRO
0,NUEVO
1,EXISTE
2,EXISTE
3,EXISTE
4,EXISTE
...,...
38244,EXISTE
38245,EXISTE
38246,EXISTE
38248,EXISTE


In [16]:
# 13 Limpieza ---- aquí se usa NumPy para cambiarlo que está mal "Sin datos" y otras cosas por NaN

cosas_mal = ["SIN DATO", "DESCONOCIDO", "???", "----"]

train_df["USO_EDIFICIO"] = train_df["USO_EDIFICIO"].replace(cosas_mal, np.nan)

train_df["TIPO_REGISTRO"] = train_df["TIPO_REGISTRO"].replace(cosas_mal, np.nan)

In [17]:
# 14 Mirada de control post-limpieza

print(train_df["USO_EDIFICIO"].unique())
print(train_df["TIPO_REGISTRO"].unique())
print(train_df["FICHERO_ORIGEN"].unique())
print(train_df["USO_EDIFICIO"].value_counts())

['VIVIENDA/S UNIFAMILIAR' 'VIVIENDA INDIVIDUAL' nan 'LOCAL COMERCIAL'
 'OTROS' 'RESTAURANTE' 'OFICINAS' 'VALENCIA'
 'EDIFICIO DE VIVIENDAS (BLOQUE COMPLETO)' 'VALÈNCIA' 'HOSPITAL'
 'EDIFICIO COMERCIAL' 'CENTRO DE ENSEÑANZA' 'HOTEL'
 'INSTALACIONES DEPORTIVAS']
['NUEVO' 'EXISTE' 'VALENCIA' nan 'VALÈNCIA' 'REFORMA']
[nan 'certificacion-energetica-de-edificios-en-alicante.csv'
 'certificacion-energetica-de-edificios-en-valencia.csv'
 'certificacion-energetica-de-edificios-en-castellon.csv']
USO_EDIFICIO
VIVIENDA INDIVIDUAL                        23409
VIVIENDA/S UNIFAMILIAR                      8674
LOCAL COMERCIAL                             1650
EDIFICIO DE VIVIENDAS (BLOQUE COMPLETO)      468
OFICINAS                                     317
OTROS                                        215
VALENCIA                                     150
EDIFICIO COMERCIAL                            79
VALÈNCIA                                      71
RESTAURANTE                                   65
CENT